### ISSUE TITLE
[ENH]  make `create_bids_filename` part of the user facing API and uniformize bids file representation

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#4172]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed enhancement in detail.

There already a BIDS filename parsing function.

https://nilearn.github.io/stable/modules/generated/nilearn.interfaces.bids.parse_bids_filename.html#nilearn.interfaces.bids.parse_bids_filename

We have a private `create_bids_filename` function.

I would suggest to make `create_bids_filename` part of the user facing API.

### Benefits to the change

Help users create output files with at least standard filenaming pattern:

```
[entity-label_]*_suffix.extension
```

and at best a bids like filenames where the filename entity-label are ordered according to bids.

### Pseudocode for the new behavior, if applicable

Ideally it would be preferable to also have a uniform representation of bids file to allow making it easier to manipulate those filenames.

```python
# create the sidecar filename for a data file
bids_file = parse_bids_filename(filename)
bids_file["extension"] = "json"
sidecar = create_bids_filename(bids_file)
```

**Comment by bthirion** [comment]
LGTM.
The only caveat [not particularly related to this issue, but ti  the general fact that we extend BIDS support withinh Nilearn] is that we may progressively reimplement a full-size BIDS handling library...

**Comment by Remi-Gau** [comment]
> LGTM. The only caveat [not particularly related to this issue, but ti the general fact that we extend BIDS support withinh Nilearn] is that we may progressively reimplement a full-size BIDS handling library...

yeah I know that's why at the moment I'd rather just make public some things that are already in the codebase: I am defo not willing to reimplement things that exist outside of nilearn.

**Comment by emdupre** [comment]
Also a little hesitant on this ; I guess I'm not sure I totally understand the use-case ? IIRC we have that function since pybids was too heavy of a dependency to add in for this functionality.

For a user, though, I'd think they're using pybids (or ancp-bids, or their favorite library) for dedicated BIDS handling. So I'm not sure when we really expect this to be used ?

**Comment by Remi-Gau** [comment]
yeah probably not required unless users ask for it
